vault/resource_aws_auth_backend_roletag_blacklist.go

```diff
@@ -0,0 +1,153 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	awsAuthBackendRoleTagBlacklistBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/config/tidy/roletag-blacklist$")
+)
+
+func awsAuthBackendRoleTagBlacklistResource() *schema.Resource {
+	return &schema.Resource{
+		Create: awsAuthBackendRoleTagBlacklistWrite,
+		Read:   awsAuthBackendRoleTagBlacklistRead,
+		Update: awsAuthBackendRoleTagBlacklistWrite,
+		Delete: awsAuthBackendRoleTagBlacklistDelete,
+		Exists: awsAuthBackendRoleTagBlacklistExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"safety_buffer": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The amount of extra time that must have passed beyond the roletag expiration, before it's removed from backend storage.",
+			},
+			"disable_periodic_tidy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
```

Comment:
Does Vault provide defaults for `disable_periodic_tidy` or `safety_buffer`? Assuming these are the docs:

- https://www.vaultproject.io/docs/auth/aws.html

It seems periodic tidying is on by default, and `safety_buffer` is by default 72. My concern here is that without defaults, once enabled or added to the configuration, removing them will have no effect if those values were set to something besides the default, e.g. `disable_periodic_tidy` is `false` by default in this context (because periodic tidying is on by default), so if a user were to mark it `true` in the configuration, apply it, then later change it to `false`, there would be no change detected by Terraform.

Reply:
It does provide defaults, and `disable_periodic_tidy` defaults to `false` and `safety_buffer` defaults to `72h`, as you've noted. So yeah, this is a bug, good catch.

I'll add a default value to both fields. This does mean overriding the Vault default value if it ever changes, but in thinking it through, I think it's a more predictable user experience than GetOkExists+Computed.